### ISSUE TITLE
Consul service doesn't work if port is omitted

### ISSUE THREAD

**Issue by phinze** [hashicorp/nomad#1084]
### Nomad version

master

### Operating system and Environment details

all

### Issue

When a service is defined without a port, nomad fails to register the service in Consul.

### Reproduction steps

Create a nomad job w/ the following service block:

```hcl
      service {
        name = "foo"
      }
```

This should work, but instead results in the following logs in the client:

```
2016/04/12 16:43:25 [ERR] driver.docker: error registering services with consul for task: "foo": 1 error(s) occurred:

* host for the service "foo"  couldn't be found
```

Investigation reveals that the code seems to assume that a `PortLabel` is specified, when this should be optional.

**Comment by diptanu** [comment]
@phinze This was by design, wasn't sure if users would want to register services which don't offer a service over the network. This can be easily fixed. We can get it out by 0.3.2

**Comment by dadgar** [comment]
Fixed by #1092
